minerl/Malmo/Minecraft/src/main/java/com/microsoft/Malmo/MissionHandlers/ObservationFromDamageImplementation.java

```diff
@@ -0,0 +1,131 @@
+package com.microsoft.Malmo.MissionHandlers;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonPrimitive;
+import com.microsoft.Malmo.MissionHandlerInterfaces.IObservationProducer;
+import com.microsoft.Malmo.Schemas.MissionInit;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.entity.EntityPlayerSP;
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.EntityLivingBase;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.DamageSource;
+import net.minecraft.util.math.Vec3d;
+import net.minecraftforge.common.MinecraftForge;
+import net.minecraftforge.event.entity.living.LivingDeathEvent;
+import net.minecraftforge.event.entity.living.LivingHurtEvent;
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;
+
+
+import static java.lang.Math.*;
+
+/**
+ * Simple IObservationProducer object that pings out a whole bunch of data.<br>
+ */
+public class ObservationFromDamageImplementation extends HandlerBase implements IObservationProducer
+{
+    private DamageSource damageSource;
+    private float damageAmount;
+    private EntityLivingBase entity;
+    private boolean hasDied = false;
+
+
+	@Override
+	public void prepare(MissionInit missionInit) {
+        MinecraftForge.EVENT_BUS.register(this);
+      }
+
+	@Override
+	public void cleanup() {}
+
+    @SubscribeEvent
+    public void onClientTick(LivingDeathEvent event)
```

Comment:
Should this be called onDeath instead of onTick?

Reply:
Huhh good catch